lib/doorkeeper/oauth/authorization/token.rb

```diff
@@ -36,7 +36,7 @@ def issue_token
             pre_auth.client,
             resource_owner.id,
             pre_auth.scopes,
-            self.class.access_token_expires_in(configuration, pre_auth),
+            self.class.access_token_expires_in(configuration, pre_auth, 'implicit'),
```

Comment:
Use constants for such things as I described in the topic comments.

Reply:
Hi @nbulaj Where to define these constants for grant flows as there is no class for grants in Doorkeeper::OAuth module. Should I define them  in BaseRequest class

Reply:
Let's place them into `Doorkeeper::Request` module as `GRANT_TYPES`

Reply:
No, let's introduce  `oauth.rb` with explicit OAuth module:

```ruby
module Doorkeeper
  module OAuth
    GRANT_TYPES = [
     # ...
    ]. freeze
  end
end
```

Reply:
I was thinking the same as It would be better to place them in separate file.